### ISSUE TITLE
Potential BUG

### ISSUE THREAD

**Issue by ramsever** [HeyLynne/FocalLoss_for_multiclass#3]
There is a bug in the focal loss function

The following line is incorrect:

 pt = label_onehot * log_p

it should be:

 pt = label_onehot * p

Where p is the probability and not the log probability

Please confirm.

**Comment by FT115** [comment]
I think you are right!
When I use log probability I get Nan loss, but when I use probability (p = F.softmax(logits)), my loss looks normal!

**Comment by HeyLynne** [comment]
Thanks. Will fix on it

**Comment by aliwaqas333** [comment]
`pt = label_onehot * torch.exp(log_p)`
